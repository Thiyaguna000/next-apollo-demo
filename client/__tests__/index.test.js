import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Home from '../pages/index';

afterEach(cleanup);

it('renders Home', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Home', () => {
    it('renders a heading', () => {
      render(<Home />)
  
      const heading = screen.getByRole('heading', {
        name: "Welcome",
      })
  
      expect(heading).toBeInTheDocument()
    })
  });

  it('Checking for Label renders in Home Component', () => {
    const {getByText} = render(
      <Home />,
    );
  
    expect(getByText("About")).toBeTruthy();
  });
