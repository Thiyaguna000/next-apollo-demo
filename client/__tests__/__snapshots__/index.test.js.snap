// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home 1`] = `
<div>
  <h1>
    Welcome
  </h1>
  <br />
  <br />
  <a
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    About
  </a>
</div>
`;
